File: frontend/src/types/lesson.ts
export type LessonStep = {
  id: string;
  conceptTitle: string; // the concept to learn in this step
  description: string;  // what this step covers
  objective: string;    // outcome the agent should achieve
  userObjective?: string; // optional user action/outcome
  done: boolean;
  order: number;
};

export type LessonPlan = {
  title: string;
  description: string;
  goal: string;
  objective: string;
  userObjective?: string;
  steps: LessonStep[];
};

export type Session = {
  id: string;
  roomId: string;
  bbSessionId: string;
  bbLiveViewUrl: string;
  bbDevtoolsWssUrl: string;
  status: "active" | "ended" | "error";
  createdAt: number;
};

File: backend/voice_bot/schemas/lesson.py
from __future__ import annotations

from typing import Optional
from pydantic import BaseModel


class LessonStep(BaseModel):
  id: str
  conceptTitle: str
  description: str
  objective: str
  userObjective: Optional[str] = None
  done: bool
  order: int


class LessonPlan(BaseModel):
  title: str
  description: str
  goal: str
  objective: str
  userObjective: Optional[str] = None
  steps: list[LessonStep]


